implementations/rust/ockam/ockam_api/tests/portals.rs

```diff
@@ -0,0 +1,628 @@
+    std::env::set_var("OCKAM_LOG", "none");
+
+    let result: ockam::Result<_> = runtime_cloned.block_on(async move {
+        let test_body = async move {
+            let echo_server_handle = start_tcp_echo_server().await;
+
+            let first_node = TestNode::create(runtime.clone(), None).await;
+            let second_node = TestNode::create(runtime, None).await;
+
+            let _outlet_status = second_node
+                .node_manager
+                .create_outlet(
+                    &second_node.context,
+                    echo_server_handle.chosen_addr,
+                    Address::from_string("outlet"),
+                    "outlet_alias".to_string(),
+                    true,
+                    Some(Arc::new(AllowAll)),
+                    None,
+                )
+                .await?;
+
+            let second_node_listen_address = second_node
+                .cli_state
+                .get_node(&second_node.node_manager.node_name())
+                .await?
+                .tcp_listener_address()
+                .unwrap();
+
+            let passthrough_server_handle =
+                start_passthrough_server(&second_node_listen_address.to_string(), disruption).await;
+
+            // create inlet in the first node pointing to the second one
+            let inlet_status = first_node
+                .node_manager
+                .create_inlet(
+                    &first_node.context,
+                    "127.0.0.1:0".to_string(),
+                    "inlet_alias".to_string(),
+                    route![],
+                    route![],
+                    InternetAddress::from(passthrough_server_handle.chosen_addr)
+                        .multi_addr()?
+                        .concat(&MultiAddr::from_str("/secure/api/service/outlet")?)?,
+                    None,
+                    None,
+                    None,
+                    true,
+                )
+                .await?;
+
+            info!("inlet_status: {inlet_status:?}");
+
+            // send 10MB of random data and verify it's correct on the other side
+            let payload_size = 10 * 1024 * 1024;
+            let mut random_buffer: Vec<u8> = vec![0; payload_size];
+            rand::thread_rng().fill_bytes(&mut random_buffer);
+
+            // connect to inlet_status.bind_addr and send dummy payload
+            let stream = TcpStream::connect(inlet_status.bind_addr.clone())
+                .await
+                .unwrap();
+
+            // we can't send and read the data from a sigle async context
+            let (mut rx, mut tx) = stream.into_split();
+
+            let copied_buffer = random_buffer.clone();
+            let _join = spawn(async move {
+                let _ = tx.write_all(&copied_buffer).await;
+                tx
+            });
+
+            let mut incoming_buffer = Vec::new();
+            let size = rx.read_to_end(&mut incoming_buffer).await.unwrap();
```

Comment:
Did I understand correctly, that we check that the connection was interrupted because otherwise this call won't return?

Reply:
@davide-baldo maybe worth adding a comment, was not obvious for me